cunumeric/module.py

```diff
@@ -1102,6 +1102,125 @@ def moveaxis(a, source, destination):
 # Changing number of dimensions
 
 
+def _reshape_recur(ndim, arr):
+    if arr.ndim < ndim:
+        cur_shape = _reshape_recur(ndim - 1, arr)
+        if ndim == 2:
+            cur_shape = (1,) + cur_shape
+        else:
+            cur_shape = cur_shape + (1,)
+    else:
+        cur_shape = arr.shape
+    return cur_shape
+
+
+def _atleast_nd(ndim, arys, view=True):
+    inputs = arys = list(convert_to_cunumeric_ndarray(arr) for arr in arys)
+    if view:
+        inputs = list(arr.view() for arr in arys)
+    result_shapes = list(_reshape_recur(ndim, arr) for arr in inputs)
+    # 'reshape' change the shape of arrays only when arr.shape != shape
+    result = list(
+        arr.reshape(shape) for arr, shape in zip(inputs, result_shapes)
```

Comment:
Just a nitpick, but can't you just call `_reshape_recur` here? Looks like you don't use those shapes again.